packages/plugin-ext/src/plugin/plugin-manager.ts

```diff
@@ -255,17 +275,29 @@ export class PluginManagerExtImpl implements PluginManagerExt, PluginManager {
             stopFn = pluginMain[plugin.lifecycle.stopMethod];
         }
         if (typeof pluginMain[plugin.lifecycle.startMethod] === 'function') {
-            const pluginExport = await pluginMain[plugin.lifecycle.startMethod].apply(getGlobal(), [pluginContext]);
-            this.activatedPlugins.set(plugin.model.id, new ActivatedPlugin(pluginContext, pluginExport, stopFn));
-
-            // resolve activation promise
-            if (this.pluginActivationPromises.has(plugin.model.id)) {
-                this.pluginActivationPromises.get(plugin.model.id)!.resolve();
-                this.pluginActivationPromises.delete(plugin.model.id);
+            try {
+                const pluginExport = await pluginMain[plugin.lifecycle.startMethod].apply(getGlobal(), [pluginContext]);
+                this.activatedPlugins.set(plugin.model.id, new ActivatedPlugin(pluginContext, pluginExport, stopFn));
+
+                // resolve activation promise
+                if (this.pluginActivationPromises.has(plugin.model.id)) {
+                    this.pluginActivationPromises.get(plugin.model.id)!.resolve();
+                    this.pluginActivationPromises.delete(plugin.model.id);
+                }
+            } catch (err) {
+                if (this.pluginActivationPromises.has(plugin.model.id)) {
+                    this.pluginActivationPromises.get(plugin.model.id)!.reject();
+                }
+                const id = plugin.model.displayName || plugin.model.id;
+                this.messageRegistryProxy.$showMessage(MainMessageType.Error, `Activating extension ${id} failed: ${err.message}.`, {}, []);
+                console.error(`Error on activation of ${plugin.model.name} - ${err}`);
+                return false;
             }
         } else {
-            console.log(`There is no ${plugin.lifecycle.startMethod} method on plugin`);
+            console.log(`There is no ${plugin.lifecycle.startMethod} method on plugin so the module is the extension's exports`);
```

Comment:
`startMethod` can be `undefined`, it means that an extension does not have any code, but it still should be activated with `pluginMain` as exports, see: https://github.com/TypeFox/vscode/blob/70b8db24a37fafc77247de7f7cb5bb0195120ed0/src/vs/workbench/api/common/extHostExtensionService.ts#L400-L401

Please also include a link in the code as a comment for future readers.

Reply:
added the link